lib/explorer/backend/lazy_series.ex

```diff
@@ -376,15 +365,36 @@ defmodule Explorer.Backend.LazySeries do
   # Returns the inner `data` if it's a lazy series. Otherwise raises an error.
   defp lazy_series!(series) do
     case series do
-      %Series{data: %__MODULE__{} = lazy} ->
-        lazy
+      %Series{data: %__MODULE__{}} ->
+        data!(series)
 
       %Series{} ->
         raise ArgumentError, "expecting a LazySeries, but instead got #{inspect(series)}"
     end
   end
 
-  defp series_or_lazy_series!(%Series{data: data}), do: data
+  defp series_or_lazy_series!(%Series{} = series), do: data!(series)
+
+  defp has_lazy_series!(items) do
+    has_lazy_series =
+      items
+      |> Enum.any?(fn
+        %Series{data: %__MODULE__{}} -> true
+        _ -> false
+      end)
+
+    case has_lazy_series do
```

Comment:
Do we need this check? If we got here, it is because at least one of them was lazy, no?

Reply:
Right! It's a defensive code, but we can remove it.

Reply:
Yes, let's remove it then! :)